internal/buffer/cursor.go

```diff
@@ -431,6 +431,127 @@ func (c *Cursor) WordLeft() {
 	c.Right()
 }
 
+// SubWordRight moves the cursor one sub-word to the right
+func (c *Cursor) SubWordRight() {
+	if (util.IsWhitespace(c.RuneUnder(c.X))) {
+		for util.IsWhitespace(c.RuneUnder(c.X)) {
+			if c.X == util.CharacterCount(c.buf.LineBytes(c.Y)) {
+				c.Right()
+				return
+			}
+			c.Right()
+		}
+		return
+	}
+	if (util.IsNoneWordChar(c.RuneUnder(c.X))) {
+		for util.IsNoneWordChar(c.RuneUnder(c.X)) {
```

Comment:
So if it is on a non-word non-whitespace character, it jumps to the nearest word or whitespace? Again, looks inconsistent with `WordRight` and `WordLeft`. But actually this seems like a reasonable behavior. But it would be reasonable for `WordRight` and `WordLeft` as well. So I'd suggest to fix `WordRight` and `WordLeft` first, and then implement `SubWordRight` and `SubWordLeft` based on that improved behavior of `WordRight` and `WordLeft`.